soc/xtensa/intel_adsp/cavs/include/intel_tgl_adsp/adsp_imr_layout.h

```diff
@@ -0,0 +1,36 @@
+/* Copyright (c) 2023 Intel Corporation
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_SOC_INTEL_ADSP_TGL_IMR_LAYOUT_H_
+#define ZEPHYR_SOC_INTEL_ADSP_TGL_IMR_LAYOUT_H_
+
+/* These structs and macros are from the ROM code header
+ * on cAVS platforms, please keep them immutable
+ */
+
+/*
+ * A magic that tells ROM to jump to imr_restore_vector instead of normal boot
+ */
+#define ADSP_IMR_MAGIC_VALUE		0x02468ACE
+#define IMR_L1_CACHE_ADDRESS		0xB0000000
```

Comment:
Nitpick: memory layout constants belong in devicetree, not C code.  Pretty sure there's already a block defined there, no?

Reply:
@andyross FYI, this cavs25 version is more-or-less copy exact from the Intel ace version  (soc/xtensa/intel_adsp/ace/include/intel_ace15_mtpm/adsp_imr_layout.h).

Reply:
now put to device tree